src/wp-includes/class-wp-query.php

```diff
@@ -5115,4 +5122,51 @@ public function lazyload_comment_meta( $check, $comment_id ) {
 		_deprecated_function( __METHOD__, '4.5.0' );
 		return $check;
 	}
+
+	/**
+	 * Optimizes ORDER BY RAND() queries using probability sampling.
+	 * Modifies the WHERE clause to include a probability condition based on RAND().
+	 * See: https://core.trac.wordpress.org/ticket/18836
+	 *
+	 * @since 6.8.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param array  $q     Query vars.
+	 * @param string $where The current WHERE clause.
+	 * @param string $join  The current JOIN clause.
+	 * @return string The potentially modified WHERE clause.
+	 */
+	protected function _optimize_rand_orderby( $q, $where, $join ) {
+		global $wpdb;
+
+		// Default factor to over-select rows, configurable via query var.
+		$rand_factor = isset( $q['rand_factor'] ) ? (float) $q['rand_factor'] : 3.0;
```

Comment:
This gets set to an empty string by default, so was resulting in `0` each time.

```suggestion
		$rand_factor = ! empty( $q['rand_factor'] ) ? (float) $q['rand_factor'] : 3.0;
```